homeassistant/components/livisi/coordinator.py

```diff
@@ -0,0 +1,135 @@
+"""Code to manage fetching LIVISI data API."""
+from __future__ import annotations
+
+from datetime import timedelta
+from typing import Any
+
+from aiolivisi import AioLivisi, LivisiEvent, Websocket
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.dispatcher import async_dispatcher_send
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+
+from .const import (
+    AVATAR_PORT,
+    CLASSIC_PORT,
+    DEVICE_POLLING_DELAY,
+    LIVISI_REACHABILITY_CHANGE,
+    LIVISI_STATE_CHANGE,
+    LOGGER,
+)
+
+
+class LivisiDataUpdateCoordinator(DataUpdateCoordinator[list[dict[str, Any]]]):
+    """Class to manage fetching LIVISI data API."""
+
+    config_entry: ConfigEntry
+
+    def __init__(
+        self, hass: HomeAssistant, config_entry: ConfigEntry, aiolivisi: AioLivisi
+    ) -> None:
+        """Initialize my coordinator."""
+        super().__init__(
+            hass,
+            LOGGER,
+            name="Livisi devices",
+            update_interval=timedelta(seconds=DEVICE_POLLING_DELAY),
+        )
+        self.config_entry = config_entry
+        self.hass = hass
+        self.aiolivisi = aiolivisi
+        self.websocket = Websocket(aiolivisi)
+        self.devices: list[str] = []
+        self.rooms: dict[str, Any] = {}
+        self.serial_number: str = ""
+        self.controller_type: str = ""
+        self.os_version: str = ""
+        self.is_avatar: bool = False
+        self.port: str = ""
+
+    async def _async_update_data(self) -> list[dict[str, Any]]:
+        """Get device configuration from LIVISI."""
+        try:
+            return await self.async_get_devices()
+        except Exception as exc:
```

Comment:
What exception do we expect? Can we narrow this down?

Reply:
We expect ClientConnectorError from aiohttp, I changed the code expecting this exception. Thank you.